consumer.go

```diff
@@ -110,9 +117,17 @@ func NewConsumerWithOptions(options *ConsumerOptions) (*Consumer, error) {
 		options.ReclaimInterval = 1 * time.Second
 	}
 
-	r, err := newRedisClient(options.RedisOptions)
-	if err != nil {
-		return nil, errors.Wrap(err, "error creating redis client")
+	var r *redis.Client
+
+	if options.RedisClient != nil {
+		r = options.RedisClient
+	} else {
+		rc, err := newRedisClient(options.RedisOptions)
```

Comment:
The `newRedisClient` function actually does a bit more than just create a new Redis client. It also checks and makes sure that the Redis version is >= v5 so that if the Redis instance doesn't support streams, it fails early on with a clear error message instead of failing later because a command doesn't exist.

I'm think that it now makes sense to split up that functionality into two functions: one that creates the Redis client and the other that verifies the version. That way, we can even verify the version for a passed-in client. We might not even need a function for creating a new Redis client, since that's only 2 lines of code. We can just do that inline here, and then after we've appropriately set `r`, we pass `r` into the function that verifies the version.

Reply:
👍  I've split this in two and adjusted the tests.